api/app/Http/Controllers/Api/DocumentationController.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use ReflectionMethod;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Route;
+use Illuminate\Foundation\Http\FormRequest;
+
+class DocumentationController extends Controller
+{
+    /**
+     * List all the routes with metadata.
+     */
+    public function index()
+    {
+        $routeCollection = Route::getRoutes();
+
+        $data = collect($routeCollection)->map(function ($route) {
+            return [
+                'url'         => $this->getRouteUri($route),
+                'description' => $this->getRouteDescription($route),
+                'methods'     => $this->getRouteMethods($route),
+                'middleware'  => $this->getRouteMiddleware($route),
+                'parameters'  => $this->getRouteParameters($route),
```

Comment:
Add 'return' values with description from Model (so all API documentation can be generated automatically)